src/components/TemplateLibrary/index.js

```diff
@@ -177,8 +177,37 @@ const LibraryComponent = () => {
     const template = await Template.fromUrl(templateIndex.ciceroUrl);
     const ciceroMark = templateToCiceroMark(template);
     const templateIdentifier = template.getIdentifier();
-    setup(ciceroMark, template);
-    saveTemplateToXml(templateIdentifier);
+    Office.context.document.customXmlParts.getByNamespaceAsync(CUSTOM_XML_NAMESPACE, result => {
```

Comment:
I think it would be a better idea to combine this with `saveTemplateToXml` as these callbacks are unnecessarily making the codebase huge.

Your code can be [put somewhere here instead](https://github.com/accordproject/cicero-word-add-in/pull/44/files#diff-4fea1789b5aec0804243466b09cef28905138a856013ec86e70e7f824035affaR221).

Reply:
@algomaster99 
Yeah, this seems to be better. I will implement these ASAP.